sw/device/lib/crypto/include/datatypes.h

```diff
@@ -79,12 +79,16 @@ typedef enum otcrypto_status_value {
  * Note: the caller must (1) allocate sufficient space and (2) set the `len`
  * field and `data` pointer when `otcrypto_byte_buf_t` is used for output. The
  * crypto library will throw an error if `len` doesn't match expectations.
+ *
+ * The order of `data` and `len` is important here for Rust compatibility; if
+ * `data` comes first, the representation exactly matches slices and is easier
+ * to call from Rust.
```

Comment:
It's possible that the current compiler implementation always puts `data` first, but the layout of slices is not guaranteed, so that's not something we can rely on.

Reply:
I see, that's a shame. I'll remove the comment.